tests/storage/sdm_copy_data_test.py

```diff
@@ -563,6 +563,49 @@ def test_copy_data_illegal(
     assert jobs.STATUS.DONE == job.status
 
 
+def test_lock_image():
+    source_img_id = str(uuid.uuid4())
+    source_vol_id = str(uuid.uuid4())
+
+    dest_vol_id = str(uuid.uuid4())
+
+    # If the endpoint is div and the image is the same, we should not lock.
+    source = dict(
+        endpoint_type='div',
+        sd_id=str(uuid.uuid4()),
+        img_id=source_img_id,
+        vol_id=source_vol_id)
+    dest = dict(
+        endpoint_type='div',
+        sd_id=str(uuid.uuid4()),
+        img_id=source_img_id,
+        vol_id=dest_vol_id)
+
+    job = copy_data.Job(str(uuid.uuid4()), 0, source, dest)
+
+    assert not job._source.lock_image
```

Comment:
Leta avoid testing private attributes. Instead test that the job succeeded, and the image was copied correctly.

We must have tests for:
- copy div to external
- copy external to div

The external endpoint can be a temporary image created by the test.

Reply:
Added test from div to external, this test fails without the change in copy_data with the same KeyError. If the approach is sound I'll the external to div test